bindings/rust/bench/benches/connection_creation.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
If you didn't already, it's probably a good idea to re-run this with a more recent allocator (e.g., jemalloc) and see how much that changes the results. Still, ~2 microsecond of difference feels to me like there is probably no reason to use a connection pool -- maybe we have plans that suggest we can get more out of pooling than we do today in this benchmark?

Reply:
Rerunning the benchmarks with [jemalloc](https://crates.io/crates/jemallocator) I got the following results.

```
Connection Creation/connection reuse
                        time:   [751.49 ns 751.64 ns 751.78 ns]
                        change: [-3.3691% -3.3445% -3.3186%] (p = 0.00 < 0.05)
                        Performance has improved.

Connection Creation/connection allocation
                        time:   [2.2163 µs 2.2170 µs 2.2178 µs]
                        change: [-23.654% -23.616% -23.581%] (p = 0.00 < 0.05)
                        Performance has improved.
```
Essentially jemalloc improved the connection allocation/free performance by ~23%.

I agree that ~2 microseconds is a relatively small improvement.

> maybe we have plans that suggest we can get more out of pooling than we do today in this benchmark?

I'm not aware of any, although we're certainly open to suggestions! My goal with this benchmark was to have _some_ empirical data to help highlight the benefit or lack thereof. 

My understanding is that connection reuse was one of the older features of s2n-tls, so perhaps the performance benefit was more significant with older platforms/allocators?

Reply:
Connection reuse is specifically intended for situations where allocation is expensive. There would be no reason to reuse connections with a modern allocator like jemalloc. Unfortunately most of our customers don't use jemalloc ;)